src/hooks/useDetectOutsideClick.ts

```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect, MouseEvent } from "react";
+
+export const useDetectOutsideClick = (
+  el: { current: HTMLElement | null },
+  initialState: boolean,
+) => {
+  const [isOpen, setIsOpen] = useState(initialState);
+
+  useEffect(() => {
+    const onClick = (e: TouchEvent | MouseEvent): void => {
```

Comment:
요렇게만 해도될듯하네요. touchEvent 발생하면 mouseEvent도 같이 발생하거든요

```suggestion
    const onClick = (e: MouseEvent): void => {
```

Reply:
https://web.dev/mobile-touchandmouse/

Reply:
감사합니다!